core/src/main/java/org/elasticsearch/rest/action/index/RestIndexAction.java

```diff
@@ -103,15 +103,9 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
             @Override
             public RestResponse buildResponse(IndexResponse response, XContentBuilder builder) throws Exception {
```

Comment:
Maybe we can let `DocWriteResponse` implement `StatusToXContent` instead of `ToXContent`? Then we can simply use `RestToXContentListener` as listener instead of an ananoumus inner class.